File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day23SafeCrackingTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day23SafeCrackingTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day23SafeCracking("/twenty_sixteen/Day23-SafeCracking-input.txt");

        assertEquals(12315, dp.getPart1Answer().intValue());
        assertEquals(479008875, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day23SafeCracking("/twenty_sixteen/Day23-SafeCracking-example#1.txt");

        assertEquals(3, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day23SafeCrackingTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day23SafeCrackingTest  extends FlatSpec with Matchers {
  "2016 Day 23 - Input" should "calculate answers" in {
    val dp = new Day23SafeCracking("/twenty_sixteen/Day23-SafeCracking-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day23SafeCracking("/twenty_sixteen/Day23-SafeCracking-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
